src/search/searchService.ts

```diff
@@ -0,0 +1,50 @@
+import { ALGOLIA_CONFIG } from '../constants'
+import { DataProvider } from '../data-providers/DataProvider'
+import { Context, Index, Lexeme } from '../types'
+import { hashContext } from '../util'
+
+/**
+ * Search Service that initializes algoliasearch and returns handlers for seraching thoughtIndex.
+ */
+const SearchService = () => {
+  const client = window.algoliaClient
+
+  if (!client) throw new Error('SearchService wasinstantiated before algolia client was registered.')
+
+  const index = client.initIndex(ALGOLIA_CONFIG.index)
```

Comment:
Does `initIndex` need to be called every time we perform a search?

Reply:
Ahh. This can go inside `initAlgoliaSearch`.